apps/gkyl_gyrokinetic_priv.h

```diff
@@ -339,25 +365,32 @@ struct gk_proj {
     struct {
       struct gkyl_array *dens; // host-side density
       struct gkyl_array *upar; // host-side upar
-      struct gkyl_array *udrift; // host-side udrift
-      struct gkyl_array *prim_moms; // host-side prim_moms 
 
-      struct gkyl_array *dens_mod; // array for correcting density
+      struct gkyl_array *prim_moms_host; // host-side prim_moms for initialization with proj_on_basis
+      struct gkyl_array *prim_moms; // prim_moms we pass to Maxwellian projection object (potentially on device)
 
-      struct gkyl_array *prim_moms_dev; // device-side prim_moms for GPU simulations
-      struct gkyl_array *dens_dev; // device-side density for GPU simulations
-      struct gkyl_dg_bin_op_mem *mem; // memory needed in correcting density
+      bool correct_all_moms; // boolean if we are correcting all the moments
 
       struct gkyl_proj_on_basis *proj_dens; // projection operator for density
       struct gkyl_proj_on_basis *proj_upar; // projection operator for upar
-      struct gkyl_proj_on_basis *proj_udrift; // projection operator for upar
       
       union {
```

Comment:
@JunoRavin @ammarhakim these multi-nested structs and unions are hard to get used to, though I could. Is this the best way to do it?